tests/test_api.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pytest
+import pytest_asyncio
+from fastapi.testclient import TestClient
+from hexkit.providers.mongodb.testutils import (  # noqa: F401; pylint: disable=unused-import
+    mongodb_fixture,
+)
+
+from covid_microservice.api.deps import get_mongodb_pcrtest_dao
+from covid_microservice.api.main import app
+from covid_microservice.core.models import PcrTest
+
+SAMPLE_URL = "/sample"
+
+
+@pytest_asyncio.fixture(name="client_with_db")
+async def fixture_client_with_db(
+    mongodb_fixture,  # noqa: F811; pylint: disable=redefined-outer-name
+):
+    """Yield a TestClient with a test db. This replaces the DAO dependency in
+    the Router instance with the test DAO produced here"""
+    dao = await mongodb_fixture.dao_factory.get_dao(
+        name="samples",
+        dto_model=PcrTest,
+        id_field="access_token",
+    )
+    app.dependency_overrides[get_mongodb_pcrtest_dao] = lambda: dao
+    yield TestClient(app)
+    app.dependency_overrides.clear()
+
+
+@pytest.mark.asyncio
+async def test_post_sample_nominal(client_with_db):
```

Comment:
This is a bit nicer to get editor and linter support:

```suggestion
async def test_post_sample_nominal(client_with_db: TestClient):
```